### PR TITLE
RFKUI-1269 : Give up after few retries

### PR DESCRIPTION
**Pull request by lokesh-b**
@alimazouchi @ChanduKempaiah-Reflektion Please Review

FYI @BrahmSingh-Reflektion @TarandeepSingh-Reflektion

### CODE REVIEW

File: pykafka/producer.py [hunk 1]
```diff
@@ -374,13 +374,22 @@ def _produce(self, message):
         :type message: `pykafka.protocol.Message`
         """
         success = False
+        retry = 0
+        max_retries = self._max_retries * 5
```
Comment (BrahmSingh-Reflektion) [inline pykafka/producer.py:378]: What is the value of self._max_retries at this point?

File: pykafka/producer.py [hunk 2]
```diff
@@ -374,13 +374,22 @@ def _produce(self, message):
...
         while not success:
             with self._update_lock:
                 leader_id = self._topic.partitions[message.partition_id].leader.id
                 if leader_id in self._owned_brokers:
                     self._owned_brokers[leader_id].enqueue(message)
                     success = True
                 else:
+                    retry = retry + 1
+                    if retry % self._max_retries == 0:
+                        log.info("<RFK> Retries exceeded limit. Updating metadata again.")
```
Comment (BrahmSingh-Reflektion) [inline pykafka/producer.py]: log.debug, if log at all.

File: pykafka/producer.py [hunk 3]
```diff
@@ -374,13 +374,22 @@ def _produce(self, message):
...
+                        self._update()
+                    elif retry > max_retries:
+                        log.error("<RFK> Retries exceeded max limit. Giving up.")
```
Comment (BrahmSingh-Reflektion) [inline pykafka/producer.py]: log.debug, if at all.

File: pykafka/producer.py (hunk 1 above)
Comment (lokesh-b) [reply pykafka/producer.py:378]: Its 3. https://github.com/Reflektion/pykafka/blob/3315d93081c7e9f2b6f857d97052fe24bc36f3ce/pykafka/producer.py#L63

File: pykafka/producer.py (hunk 3 above)
Comment (lokesh-b) [reply pykafka/producer.py]: I can change the other one to debug if needed, but i will keep this as error. We need a way to know if this is happening a lot. Even pykafka does the same when they fail to deliver a message
https://github.com/Reflektion/pykafka/blob/3315d93081c7e9f2b6f857d97052fe24bc36f3ce/pykafka/producer.py#L487

**Pull request merged**